docs/adr/0016-publishing-times-and-change-history.md

```diff
@@ -0,0 +1,208 @@
+# 16. Publishing times and change history
+
+Date: 2020-03-20
+
+## Context
+
+Content that is published on GOV.UK has metadata to indicate the history of
+the content. This is comprised of a time to indicate when the first iteration of
+that content was published (`first_published_at`), a time for when the content
+was last changed in a meaningful way (`public_updated_at`) and a
+change history which records a change note and the publishing time of each of
+these meaningful changes. By convention, the first item in a change history of
+GOV.UK content is provided by the publishing application without publisher
+input. This item has a change note of "First published." and has the time the
+content was initially published on GOV.UK.
+
+To set these metadata fields Content Publisher has relied on the Publishing API
+automatically setting the time fields and using it's
+[ChangeNote][pub-api-change-note] model. This model stores a record for each
+major change published on GOV.UK and can collate them to build a change
+history. This meant that Content Publisher has not needed to store
+`first_published_at`, `public_updated_at` or a full change history and has only
+needed to store a change note for an edition.
+
+We had concerns that continued usage of the Publishing API time and change
+history system would lead to discrepancies when migrating content from Whitehall
+to Content Publisher. This led to us doing a review of the system and problems,
+where some relate to migration and others are broader.
+
+### Problems identified
+
+#### 1. Having accurate times for Whitehall migrated content
+
+When content is migrated from Whitehall to Content Publisher we want all the
+times associated with that content to remain the same. Whitehall complicates
+this by passing two time values to represent the first publishing time: the
+aforementioned `first_published_at` field and a deprecated time value,
+`first_public_at`, which is used [preferentially to
+`first_published_at`][first-public-at] when rendering the content. Content
+Publisher does not populate the `first_public_at` field.
+
+An occasional problem is that the Publishing API can have different values for
```

Comment:
This could do with some more explanation, as at first it conflicts with the previous statement that Whitehall is "passing two time values to represent the first publishing time", which suggests they should always be the same. I'm guessing there's a way they can differ when they're set by Whitehall, and that's the problem?

Reply:
Sure, I've updated it a bit. I think the reason for the disparity is mostly wrapped in historical issues (Whitehall didn't used to populate one, now does; Publishing API had tons of date problems) and then lingering issues of there being two fields there is scope for them not to match.